src/Orckestra.Composer.Search/Services/CategoryBrowsingViewService.cs

```diff
@@ -73,15 +71,72 @@ public virtual async Task<CategoryBrowsingViewModel> GetCategoryBrowsingViewMode
             {
                 CategoryId = param.CategoryId,
                 CategoryName = param.CategoryName,
-                SelectedFacets = await GetSelectedFacetsAsync(param).ConfigureAwait(false),
+                FacetSettings = new FacetSettingsViewModel()
+                {
+                    SelectedFacets = await GetSelectedFacetsAsync(param).ConfigureAwait(false),
+                },
                 ProductSearchResults = await GetProductSearchResultsAsync(param).ConfigureAwait(false),
-                ChildCategories = await GetChildCategoriesAsync(param).ConfigureAwait(false),
                 LandingPageUrls = landingPageUrls
             };
 
+            viewModel.FacetSettings.CategoryFacetValuesTree = await BuildCategoryFacetValuesTree(viewModel.ProductSearchResults.Facets,
+                viewModel.FacetSettings.SelectedFacets,
+                viewModel.ProductSearchResults.CategoryFacetCounts).ConfigureAwait(false);
+            if (viewModel.FacetSettings.CategoryFacetValuesTree != null)
+            {
+                var categoryRoot = CategoryRootNode(viewModel.FacetSettings.CategoryFacetValuesTree.ChildNodes, param.CategoryId);
+                viewModel.FacetSettings.CategoryFacetValuesTree.TotalCount = categoryRoot.Quantity;
+                viewModel.FacetSettings.CategoryFacetValuesTree.ChildNodes = categoryRoot.ChildNodes;
+                viewModel.FacetSettings.CategoryFacetValuesTree.ChildNodes?.ForEach(childNode => BuildCategoryUrlsForTreeNode(param, childNode));
+                viewModel.FacetSettings.CategoryFacetValuesTree.ChildNodes?.ForEach(childNode => CleanSiblingFacets(param, childNode));
+                viewModel.FacetSettings.Context["CategoryFacetValuesTree"] = viewModel.FacetSettings.CategoryFacetValuesTree;
+            }
+
+            // Json context for Facets
+            viewModel.FacetSettings.Context["SelectedFacets"] = viewModel.FacetSettings.SelectedFacets;
+            viewModel.FacetSettings.Context["Facets"] = viewModel.ProductSearchResults.Facets.Where(f => !f.FieldName.StartsWith(SearchConfiguration.CategoryFacetFiledNamePrefix));
+            viewModel.FacetSettings.Context["PromotedFacetValues"] = viewModel.ProductSearchResults.PromotedFacetValues;
+
             return viewModel;
         }
 
+        private CategoryFacetValuesTreeNode CategoryRootNode(List<CategoryFacetValuesTreeNode> nodes, string categoryId)
+        {
+            if (nodes == null) return null;
+            var rootCategory = nodes.FirstOrDefault(c => c.CategoryId == categoryId);
+            if (rootCategory == null)
+            {
+                foreach (var childNode in nodes)
+                {
+                    rootCategory = CategoryRootNode(childNode.ChildNodes, categoryId);
```

Comment:
Smth wrong, by code it looks like root category assigned from latest  childNode, and all previous calculation useless

Reply:
This is specific case for Category Page. By default we have the whole tree for Categories. 
In CategoryBrowsingViewService we rewrite the root of tree and set it to the Category page Root.
First client wanted to render all tree from 1 level root for category page. Then they changed mind to have tree started from Category page category.
Other calculations for Category Urls we do for the new rewrited Tree.